handler-notify-developer/src/main/java/pl/brightinventions/slf4android/handler/notifydeveloper/NotifyDeveloperDialogDisplayActivity.java

```diff
@@ -67,25 +70,31 @@ private static AlertDialog showDialogIn(final Context activityContext, final Str
                     public void onClick(DialogInterface dialog, int which) {
                         sendEmailWithError(activityContext, emailErrorReport);
                         dialog.dismiss();
-                        if (onDialogClose != null) {
-                            onDialogClose.dispose();
-                        }
+                        closeSafely(onDialogClose);
                     }
                 })
                 .setNegativeButton(android.R.string.no, new DialogInterface.OnClickListener() {
                     @Override
                     public void onClick(DialogInterface dialog, int which) {
                         dialog.dismiss();
-                        if (onDialogClose != null) {
-                            onDialogClose.dispose();
-                        }
+                        closeSafely(onDialogClose);
                     }
                 })
                 .create();
         alertDialog.show();
         return alertDialog;
     }
 
+    private static void closeSafely(Closeable closeable) {
+        if (closeable != null) {
+            try {
+                closeable.close();
+            } catch (IOException e) {
+                Log.e(TAG, "Could not close " + closeable, e);
```

Comment:
`Log.e` is used for consistency with the existing code in this class. Nevertheless, I don't see a reason why SLF4J API shouldn't be used here, so we might as well migrate all `Log.e` to `Logger.error`.